cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -65,16 +62,20 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
                                   @Nullable @QueryParam("application") String application,
                                   @Nullable @QueryParam("programType") @DefaultValue("workflows") String programType,
                                   @Nullable @QueryParam("programId") @DefaultValue("DataPipelineWorkflow")
-                                      String programName,
-                                  @Nullable @QueryParam("run") String run,
+                                    String programName, @Nullable @QueryParam("run") String run,
                                   @Nullable @QueryParam("maxRunsPerProgram") @DefaultValue("1")
-                                      Integer maxRunsPerProgram) throws Exception {
+                                    Integer maxRunsPerProgram) throws Exception {
     // Establishes the support bundle configuration
     SupportBundleConfiguration bundleConfig =
       new SupportBundleConfiguration(namespace, application, run, ProgramType.valueOfCategoryName(programType),
                                      programName, maxRunsPerProgram);
-    // Generates support bundle and returns with uuid
-    String uuid = bundleService.generateSupportBundle(bundleConfig);
-    responder.sendString(HttpResponseStatus.OK, uuid);
+    String status = bundleService.ensurePreviousExecutorFinish();
+    if (status == null) {
+      // Generates support bundle and returns with uuid
+      String uuid = bundleService.generateSupportBundle(bundleConfig);
+      responder.sendString(HttpResponseStatus.CREATED, uuid);
+    } else {
+      responder.sendString(HttpResponseStatus.OK, status);
```

Comment:
Please return UUID in both cases. The caller should not need to differentiate by status. If it wants, it can differentiate, but overall for any 2xx status it should get UUID